src/resmom/linux/alps.c

```diff
@@ -3727,21 +3727,22 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 						goto bad_vnl;
 					if (accel->data.gpu->memory) {
 						attr = "resources_available.accelerator_memory";
-						if (totseg == 0) {
+						if (totseg == 0 || vnode_per_numa_node) {
 							snprintf(utilBuffer,
 								sizeof(utilBuffer),
 								"%umb",
 								accel->data.gpu->memory);
-						} else if (vnode_per_numa_node) {
-							snprintf(utilBuffer,
-								sizeof(utilBuffer),
-								"@%s_%ld_0",
-								mpphost, node->node_id);
+							if (vnode_per_numa_node && !totseg) {
+								snprintf(utilBuffer,
+									 sizeof(utilBuffer),
+									 "@%s_%ld_0",
+									 mpphost, node->node_id);
+							}
+							if (vn_addvnr(nv, vname, attr,
+								      utilBuffer, 0,
+								      0, NULL) == -1)
+								goto bad_vnl;
```

Comment:
Hi Lisa,
 You were right, the previously existing code seems much cleaner and was correct. I have made the necessary changes. Please review the latest code changes.